app/controllers/api/v1/books_controller.rb

```diff
@@ -0,0 +1,87 @@
+
+module Api
+  module V1
+    class BooksController < ApplicationController
+      respond_to :json
+
+      def index
+        @q = Book.ransack(params[:q])
+
+        # Dynamically handle filtering based on provided parameters
+        books = if params[:q].present?
+          categories = params[:q][:categories_in].split(',') if params[:q][:categories_in].present?
+
+          filtered_books = @q.result.includes(:categories, :shelf)
+
+          # Filter books by multiple categories if categories parameter is provided
+          filtered_books = filtered_books.joins(:categories).merge(Category.where(id: categories)) if categories.present?
```

Comment:
using inner joins will cause to not return all books of filtered books when null value
so we should switch to left join that includes all rows from the filtered_books